Cython/Utility/CppConvert.pyx

```diff
@@ -52,14 +52,24 @@ cdef vector[X] {{cname}}(object o) except *:
 #################### vector.to_py ####################
 
 cdef extern from *:
-    cdef cppclass vector "const std::vector" [T]:
+    cdef cppclass vector "std::vector" [T]:
         size_t size()
         T& operator[](size_t)
 
-@cname("{{cname}}")
-cdef object {{cname}}(vector[X]& v):
-    return [v[i] for i in range(v.size())]
+cdef extern from "Python.h":
+    void Py_INCREF(object)
+    list PyList_New(Py_ssize_t size)
+    void PyList_SET_ITEM(object list, Py_ssize_t i, object o)
 
+@cname("{{cname}}")
+cdef object {{cname}}(const vector[X]& v):
+    o = PyList_New(<Py_ssize_t>v.size())
+    cdef Py_ssize_t i = 0
+    for i in range(v.size()):
```

Comment:
Useless assignment.
```suggestion
    cdef Py_ssize_t i
    for i in range(v.size()):
```

Reply:
Is it preferred to use
```
    cdef Py_ssize_t i
    for i in range(v.size()):
```
or
```
    for i in range(<Py_ssize_t>v.size()):
```
and let cython deduce the type of i?

Reply:
The first. It's usually better if utility code doesn't depend too much on features that users can influence. Keeps it less brittle.